src/dcap_provider.cpp

```diff
@@ -203,34 +205,44 @@ static inline quote3_error_t fill_qpl_string_buffer(
 //
 // Determine time cache should invalidate for given collateral
 //
-bool get_cache_expiration_time(CollateralTypes collateral_type, time_t &expiration_time)
+bool get_cache_expiration_time(const string &cache_control, const string &url, time_t &expiration_time)
 {
     time_t max_age = 0;
     tm* max_age_s = localtime(&max_age);
-
-    switch(collateral_type)
+    string match = "max-age=";
+    size_t index = cache_control.find(match);
+    int cache_time_seconds = 0;
+    if (index != string::npos)
     {
-        case CollateralTypes::TcbInfo:

```

Comment:
What happens if you don't have a cache_control header value? I think it means that going to immediately expire the collateral, is that the intent (and if it is, is that wise?)

Reply:
That is the intent - idea being if caching service is broken and handing out expired collateral and not populating the cache control header, azdcap won't cache anything so whatever mitigation we apply on the cache service will become active ASAP from the customer perspective without them having to do anything.

Alternative would be to cache for some predefined time.... I'm open to that as well. 24 hours seems like too long - maybe 8 hours?

Reply:
I'm ok with any behavior, but it should be explicit.

Reply:
OK I made this explicit in the log message - "Collateral will not be cached"